app/code/Magento/Catalog/Model/Product/Gallery/GalleryManagement.php

```diff
@@ -61,6 +61,10 @@ public function create($sku, ProductAttributeMediaGalleryEntryInterface $entry)
         $existingMediaGalleryEntries = $product->getMediaGalleryEntries();
         $existingEntryIds = [];
         if ($existingMediaGalleryEntries == null) {
+            // set all media types if not specified
+            if ($entry->getTypes() == null) {
```

Comment:
Please, use the strict types comparison (`===`).

Reply:
I uses loose comparison here in order to handle few cases for `$entry->getTypes()`: NULL, [] in the same way like on line 63.
Please clarify in case the extra changes needed here, thanks.